### PR TITLE
Hki 237/legend donutchart hover fix

### PR DESCRIPTION
**Pull request by hki-237**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes
Updating correct operator and adding return value

### CODE REVIEW

File: packages/charting/src/components/DonutChart/DonutChart.base.tsx [hunk 1]
```diff
@@ -265,12 +265,13 @@ export class DonutChartBase extends React.Component<IDonutChartProps, IDonutChar
   }
 
   private _valueInsideDonut(valueInsideDonut: string | number | undefined, data: IChartDataPoint[]) {
```
Comment (gloria-e-ho) [inline packages/charting/src/components/DonutChart/DonutChart.base.tsx:267]: I saw in a previous commit that when calling `this._valueInsideDonut`, we used a non-null assertion on the `valueInsideDonut` variable. If that's the case, does ` | undefined` still need to be added here?

File: packages/charting/src/components/DonutChart/DonutChart.base.tsx [hunk 2]
```diff
@@ -265,12 +265,13 @@ export class DonutChartBase extends React.Component<IDonutChartProps, IDonutChar
...
-    if (valueInsideDonut != undefined && this.state.activeLegend != '' && !this.state.showHover) {
+    if (valueInsideDonut !== undefined && this.state.activeLegend !== '' && !this.state.showHover) {
```
Comment (gloria-e-ho) [inline packages/charting/src/components/DonutChart/DonutChart.base.tsx:268]: A stylistic choice, and I know we don't own this repo so lemme know if there are differing opinions. I've begun falling in love with early return methods, which take the "simple return case" and have that be the first check, & return if it's true. In this case, that would look like changing this if-condition to 
`if (valueInsideDonut === undefined && this.state.activeLegend === '' && !this.state.showHover) {
      return legendValue;
}` 

And then we can also eliminate the `else` below it, and un-indent the remaining logic. I think code that sticks as far-left to the file as possible is the easiest to read :) 

This is a big preference thing though, so lemme know if you have other thoughts!

File: packages/charting/src/components/DonutChart/DonutChart.base.tsx [hunk 3]
```diff
@@ -265,12 +265,13 @@ export class DonutChartBase extends React.Component<IDonutChartProps, IDonutChar
...
       let legendValue = valueInsideDonut;
       data!.map((point: IChartDataPoint, index: number) => {
```
Comment (gloria-e-ho) [inline packages/charting/src/components/DonutChart/DonutChart.base.tsx:270]: Generally I think we should be careful with non-null assertions. Are we sure this operation will always be safe? If we're not certain, a good alternative would be to check to see if `data` is either `null` or `size = 0`, and then return early

File: packages/charting/src/components/DonutChart/DonutChart.base.tsx [hunk 4]
```diff
@@ -265,12 +265,13 @@ export class DonutChartBase extends React.Component<IDonutChartProps, IDonutChar
...
-        if (point.legend == this.state.activeLegend) {
+        if (point.legend === this.state.activeLegend) {
           legendValue = point.data!;
```
Comment (gloria-e-ho) [inline packages/charting/src/components/DonutChart/DonutChart.base.tsx:272]: Same here, are we certain this is a safe operation? Are there situations where `point.data` could be `null | undefined` and we're forcing a value here? And if yes, then what behavior will we see the chart exhibiting?

File: packages/charting/src/components/DonutChart/DonutChart.base.tsx (hunk 2 above)
Comment (hki-237) [reply packages/charting/src/components/DonutChart/DonutChart.base.tsx:268]: I do agree that this does visually look better. I am just maintaining the same pattern I am seeing in their codebase.

**Pull request merged**